### PR TITLE
Use precompiled handlebars

### PR DESCRIPTION
**Pull request by sharidas**
This change will help to use
precompiled handlebars.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,11 +1,13 @@
 SHELL := /bin/bash
 
+HANDLEBARS=$(CURDIR)/../../build/node_modules/handlebars/bin/handlebars
```
Comment (PVince81) [inline Makefile]: Won't work in all scenarios as it assumes that core is also from git.

If you use core tarballs with git impersonate app, this won't run.
Best is to do it like the customgroups app and grab your own handlebars version...

Comment (sharidas) [inline Makefile]: Ok, so that's why customgroups is having own handlebars. Will do that way then. Thanks for the pointer.

**Comment by felixboehm** [comment]
@sharidas what needs to be done here, to get this finished?
Or is this already outdated?

**Comment by sharidas** [comment]
I have rebased this change on top of branch: logout-logic-dependon-prehooks ( essentially its on top of the PR: https://github.com/owncloud/impersonate/pull/68).

File: Makefile [hunk 2]
```diff
@@ -19,7 +27,19 @@ endif
 # Catch-all rules
 #
 .PHONY: dist
-dist: build-src
+dist: build-dep js-templates build-src
+
+%.js: $(template_src)
+		$(HANDLEBARS) $* -f $@
```
Comment (PVince81) [inline Makefile]: please add a namespace, we don't want the ones from different apps to collide

**Comment by sharidas** [comment]
Rebased this PR on top of the master branch.

**Pull request merged**